src/main/java/net/geforcemods/securitycraft/compat/waila/JadeDataProvider.java

```diff
@@ -52,66 +50,37 @@
 
 @WailaPlugin(SecurityCraft.MODID)
 public class JadeDataProvider extends WailaCompatConstants implements IWailaPlugin {
-	public static final SpoofBlockIcon SPOOF_BLOCK_ICON = new SpoofBlockIcon();
 	public static final SecurityCraftInfo SECURITYCRAFT_INFO = new SecurityCraftInfo();
-	public static final SpoofBlockName SPOOF_BLOCK_NAME = new SpoofBlockName();
-	public static final SpoofModName SPOOF_MOD_NAME = new SpoofModName();
-	private final List<Class<?>> isRegistered = new ArrayList<>();
 
 	@Override
+	@OnlyIn(Dist.CLIENT)
 	public void registerClient(IWailaClientRegistration registration) {
 		registration.addConfig(SHOW_OWNER, true);
 		registration.addConfig(SHOW_MODULES, true);
 		registration.addConfig(SHOW_PASSWORDS, true);
 		registration.addConfig(SHOW_CUSTOM_NAME, true);
 
-		for (RegistryObject<Block> registryObject : SCContent.BLOCKS.getEntries()) {
-			Block block = registryObject.get();
+		registration.registerBlockComponent(SECURITYCRAFT_INFO, Block.class);
+		registration.registerEntityComponent(SECURITYCRAFT_INFO, Sentry.class);
 
-			if (!(block instanceof OwnableBlock) && !Utils.getRegistryName(block).getPath().matches("(?!(reinforced_)).*?crystal_.*") && !(block instanceof ReinforcedCauldronBlock) && !(block instanceof ReinforcedPaneBlock)) {
-				if (!isRegistered.contains(block.getClass())) {
-					registration.registerBlockComponent(SECURITYCRAFT_INFO, block.getClass());
-					isRegistered.add(block.getClass());
+		registration.addBeforeRenderCallback((tooltip, rect, poseStack, accessor, color) -> ClientHandler.isPlayerMountedOnCamera());
+		registration.addRayTraceCallback((hit, accessor, original) -> {
+			if (accessor instanceof BlockAccessor blockAccessor) {
+				if (blockAccessor.getBlock() instanceof IOverlayDisplay block) {
+					ItemStack fake = block.getDisplayStack(blockAccessor.getLevel(), blockAccessor.getBlockState(), blockAccessor.getPosition());
+					return registration.blockAccessor().from(blockAccessor).fakeBlock(fake).build();
+				}
+				if (blockAccessor.getBlock() instanceof FakeWaterBlock) {
+					BlockState camo = Blocks.WATER.defaultBlockState();
+					return registration.blockAccessor().from(blockAccessor).blockState(camo).build();
+				}
+				if (blockAccessor.getBlock() instanceof FakeLavaBlock) {
+					BlockState camo = Blocks.LAVA.defaultBlockState();
+					return registration.blockAccessor().from(blockAccessor).blockState(camo).build();
 				}
```

Comment:
else if instead of if, please. Also the variables can be inlined, and the {} removed.